domain/src/main/java/org/oppia/domain/exploration/ExplorationProgress.kt

```diff
@@ -0,0 +1,85 @@
+package org.oppia.domain.exploration
+
+import org.oppia.app.model.Exploration
+import org.oppia.app.model.State
+import org.oppia.domain.state.StateDeck
+import org.oppia.domain.state.StateGraph
+
+// TODO(#186): Use an interaction repository to retrieve whether a specific ID corresponds to a terminal interaction.
+private const val TERMINAL_INTERACTION_ID = "EndExploration"
```

Comment:
I see your comment but I just wanted to clarify what it means - you mean we should keep (something like) a map of interaction Id names as we get in the json to (something like) an enum of interaction types?

Reply:
I like that. We probably still want to have a notion of terminality, but the enum approach makes the whole management of interactions a lot cleaner. I added a note in #186.